### ISSUE TITLE
lack of interest?

### ISSUE THREAD

**Issue by turric4n** [Relfos/TERRA-Engine#14]
Hi Relfos,

Do you left Terra-Engine? You lack of interest?

**Comment by Relfos** [comment]
Hi, I still have interest in engines but I no longer use Delphi / Free Pascal, I've moved to C# a long time ago and I've been porting pieces of TERRA to C# whenever I have some free time.

**Comment by turric4n** [comment]
Delphi/Free Pascal has improved a lot... I use C# too for work (It's quite similar to Delphi) but when I arrive home I need to be in a native world again to be happy :)

**Comment by Relfos** [comment]
Both languages were created by the same guy, for me C# was an evolution over Delphi. And TERRA is more than 10 years old by now, so even I would return to it I could never resist rewriting it from scratch, as my programming style and knowledge has improved. 

However I think TERRA has lots of useful stuff here, some of those stuff is even rare to find in generally, not just in Delphi, like the pure Pascal OGG decoder or TTF loader (that I'm currently porting to C#).
If someone uses parts of this as base to their own engine or game I will be happy.

**Comment by turric4n** [comment]
Thanks Relfos, I want to use your engine because seems awesome for me. So I want to help you to mantain if you want. Don't left Pascal because needs people like you!

**Comment by Relfos** [comment]
If you want to give an hand then sure, try first using the engine to familiarize yourself with it, do you want to make a game or something else?

**Comment by turric4n** [comment]
Yes, basically I want to use this engine to make simple game with OpenGL acceleration. But I know your feelings about to rewrite the engine from scratch again and I encourage you to do so.

**Comment by Relfos** [comment]
You can use this engine right now as it is, it was used in multiple released games, one of is was released in Steam, others in Google Play Store.

The engine is fairly complete, and offers good performance for 2D and 3D games. There are also a good amount of samples showcasing the engine features, however some of them might not compile with the master branch version

**Comment by rossi42** [comment]
Hey turric4n, did you manage to get anything running under win64 with a recent compiler? I tried the new master branch and the old-master branch and both fail. I get to compile the samples after setting target to default and in the new master i needed to make cosmetic changes to the jpg part. But anyhow on both branches the executables throw a access violation.
My setup: Win10/64, lazarus-2.0.2-fpc-3.0.4-win64.

**Comment by turric4n** [comment]
Didn't test at all... Would be nice to continue project but I can't do it alone.

**Comment by Relfos** [comment]
@rossi42 FPC 3.0 and forward completely broke the engine somehow, could be the new Unicode code, or something else, I'm not sure exactly, but it was one of the reasons I lost interest in continuing working on it, when necessary to compile something I used FPC 2.7

**Comment by rossi42** [comment]
I tried it with lazarus-1.4.4-fpc-2.6.4-win64 now and got same result. 
I tried text_render demo and it got to here:

  Result._Instance := TargetClass.Create();

Which is line 640 in TERRA_Application and then causes sigsegv/runtime 216
I could try 32bit version of lazarus...

**Comment by Relfos** [comment]
Yes, probably currently only works for 32-bit, this engine was started
around 15 years ago or so.
There was an huge effort to support 64 bits already, but some code might
not work yet at all, specially the text rendering, which was based on an
old version of stb_ttf, which was ported from C to Pascal around 9 years
ago.

So while TERRA Engine surely packs lots of features, I would consider this
a legacy project at this point, would require tons of work to bring it back
to shape. I think Castle Engine is still being developed and it is
open-source I think? If yes, thats probably the best thing around for
people who want to start new games using Delphi or Free Pascal.

**Comment by rossi42** [comment]
Yeah I had a look at castle engine but it lacks networking support and that is what I was hoping for with TERRA. Well time to move on and keep searching.

**Comment by Relfos** [comment]
Try finding an lib just for networking and combine it with Castle Engine, I
don't remember the name now but there were 1 or 2 good network pascal libs,
I think that will be the best thing you can get, I dont think theres
anything at the level of Castle Engine that is still being updated.

**Comment by CodingMadness** [comment]
@Relfos So you would say, at this point ,you completly stopped working on TERRA Engine??

**Comment by rossi42** [comment]
Btw. I moved to Unity meanwhile. Got all the whistles and great community and especially for starting in the business the best commercial plan. (Free until you earn). Projects like TERRA are super important for the community and 15 years ago when this project started it sure was something unique. Therefore: thank you very much Relfos for providing this!
Make code not war. Peace!

**Comment by Relfos** [comment]
Hello @rossi42 . I too moved to Unity and C# and due to that I did not update TERRA anymore.
It's sad but with the current state and requirements of game development in 2021, I think its no longer viable for a single person to work on a home-made game engine. I now prefer to spend my time making games instead :)
